File: src/v1/models/user.model.js
const { Schema, model } = require('mongoose');

const COLLECTION_NAME = 'Users';
const DocumentName = 'User';

const userSchema = new Schema({
    username: { type: String, required: true , unique: true},
    password: { type: String, required: true },
    email: { type: String, required: true, unique: true },
    phone: { type: String, },
    address: { type: String },
    fullname: { type: String },
    avatar: { type: String },
    images: { type: Array, default: []},
    status: { type: String, enum: ['active', 'inactive'], default: 'active' },
    ordering: { type: Number, default: 0 },
    role: { type: String, enum: ['admin', 'user'], default: 'user' },
    resetPasswordToken: String,
    resetPasswordExpires: Date,
}, {
    timestamps: true,
    collection: COLLECTION_NAME,
});

userSchema.index({ username: 1, email: 1,  });

module.exports = model(DocumentName, userSchema);

File: lib/models/user.model.dart
class User {
  final String id;
  final String username;
  final String email;
  final String phone;
  final String? address;
  final String? fullname;
  final String? avatar;
  String status;
  final int ordering;
  final String role;

  User({
    required this.id,
    required this.username,
    required this.email,
    required this.phone,
    this.address,
    this.fullname,
    this.avatar,
    required this.status,
    required this.ordering,
    required this.role,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['_id'] as String? ?? '',
      username: json['username'] as String? ?? '',
      email: json['email'] as String? ?? '',
      phone: json['phone'] as String? ?? '',
      address: json['address'] as String?,
      fullname: json['fullname'] as String?,
      avatar: json['avatar'] as String?,
      status: json['status'] as String? ?? 'active',
      ordering: json['ordering'] as int? ?? 0,
      role: json['role'] as String? ?? 'user',
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'username': username,
        'email': email,
        'phone': phone,
        'address': address,
        'fullname': fullname,
        'avatar': avatar,
        'status': status,
        'ordering': ordering,
        'role': role,
      };
}
